### ISSUE TITLE
[BUG] Package is incompatible with FlutterFlow v4.1

### ISSUE THREAD

**Issue by davyskiba** [nomtek/mistralai_client_dart#40]
**Describe the bug**
`mistralai_client_dart` in version 0.1.0 is using dependencies incompatible with FlutterFlow v4.1:
- Flutter SDK 3.1.3 used by FlutterFlow does not fit in the required version range >=3.2.0 <4.0.0
- http dependency 0.13.0 used indirectly by FlutterFlow does not match http 1.2.0 used by `mistralai_client_dart`.

**To Reproduce**
1. Create an empty new flutter flow project
2. Add custom action depending on `mistralai_client_dart`
3. Try to run the project either on the web or locally

**Expected behavior**
Version resolving completes and project runs

**Additional context**

```
Because test_project depends on mistralai_client_dart >=0.0.2 which requires SDK version >=3.2.0 <4.0.0, version solving failed.
```

```
Because no versions of mistralai_client_dart match >0.1.0 <0.2.0 and mistralai_client_dart 0.1.0 depends on http ^1.2.0, mistralai_client_dart ^0.1.0 requires http ^1.2.0.
And because google_fonts 4.0.3 depends on http ^0.13.0, mistralai_client_dart ^0.1.0 is incompatible with google_fonts 4.0.3.
So, because test_project depends on both google_fonts 4.0.3 and mistralai_client_dart ^0.1.0, version solving failed.
```

**Comment by gawi151** [comment]
Ok so to make this happen we would have to create a special version of package on pub.dev for FlutterFlow. It's due to old dependencies FlutterFlow is using.

I created an issue about unsafe package version they are using here https://github.com/FlutterFlow/flutterflow-ui/issues/17.

So plan would be to:
- create seperate branch for FlutterFlow version of the package
- downgrade all necessary package/sdk versions so it's compatible with FlutterFlow
- test it with current version of FlutterFlow
- update README about support for FlutterFlow
- publish new package to pub.dev

**Comment by gawi151** [comment]
Resolved in #41
